### PR TITLE
Implement `Withdraw` in the client.

### PR DESCRIPTION
**Pull request by enriquefynn**
Just as the `Deposit` instruction has a client-side implementation, this PR implements the client-side for `Withdraw`.
It's mainly useful for testing, as withdrawals will be done in the browser UI.

### CODE REVIEW

File: cli/src/helpers.rs [hunk 1]
```diff
@@ -606,3 +614,88 @@ pub fn command_deposit(
...
+            &config.signer.pubkey(),
+            &solido.st_sol_mint,
+        );
+
+        let stake_authority =
+            solido.get_stake_authority(opts.solido_program_id(), opts.solido_address())?;
+
+        // Get heaviest validator.
+        let heaviest_validator = get_validator_to_withdraw(&solido.validators).map_err(|_| {
+            MaintenanceError::new(
```
Comment (ruuda) [inline cli/src/helpers.rs:658]: It’s not really a maintenance error. I added a catch-all `CliError` in #343 that we can use for this. If you want to merge first you can also add a
```rs
// TODO(ruuda)
```
comment here and then I’ll change it as part of #343.

File: program/src/error.rs [hunk 2]
```diff
@@ -146,6 +146,10 @@ pub enum LidoError {
     /// There is a validator that has more stake than the selected one.
     #[error("ValidatorWithMoreStakeExists")]
     ValidatorWithMoreStakeExists = 37,
+
+    /// The instance has no validators.
+    #[error("EmptySetOfValidators")]
+    EmptySetOfValidators = 38,
```
Comment (ruuda) [inline program/src/error.rs:152]: We already have `NoActiveValidators = 27`. That name is outdated, we don’t need an “active” stake account for the validator, `EmptySetOfValidators` is a better name. But we should merge the two.

**Review by ruuda — APPROVED** [review]
LGTM with two comments.

File: program/src/error.rs (hunk 2 above)
Comment (enriquefynn) [reply program/src/error.rs:152]: Great, changed it to `EmptySetOfValidators`

**Pull request merged**